test/unit_test/cats_tree_tests.cc

```diff
@@ -493,12 +493,16 @@ BOOST_AUTO_TEST_CASE(build_min_depth_tree_cont_1)
   BOOST_CHECK_EQUAL_COLLECTIONS(tree.nodes.begin(), tree.nodes.end(), expected.begin(), expected.end());
 }
 
-BOOST_AUTO_TEST_CASE(build_min_depth_tree_cont_too_big, *boost::unit_test::disabled())
-{
-  VW::cats_tree::min_depth_binary_tree tree;
-  // Throws vw_exception when unable to allocate enough memory to build tree
-  BOOST_CHECK_THROW(tree.build_tree(INT_MAX, 0), VW::vw_exception);
-}
+// This test has to be disabled for the following reasons
+// 1) Valgrind cannot handle exceptions while allocating memory
```

Comment:
https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/test/unit_test/vwdll_test.cc#L80

That might be helpful, in case you want to run the test but skip it only under the valgrind run.

Reply:
Nice!  Used this.